x/slashing/abci.go

```diff
@@ -56,6 +56,55 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 		if writeInfo == nil {
 			panic("Expected slashing write info to be non-nil")
 		}
+		// Check if we need to resize the array if there was recently a change in slashing window size
+		window := k.SignedBlocksWindow(ctx)
+		missedInfo, found := k.GetValidatorMissedBlocks(ctx, writeInfo.ConsAddr)
+		missedBlockLen := int64(len(missedInfo.MissedBlocks))
+		if found && window != missedBlockLen {
+			// we need to resize the missed block array AND update the signing info accordingly
+			switch {
+			case missedBlockLen < window:
+				// missed block array too short, lets expand it
+				indexOffset := writeInfo.SigningInfo.IndexOffset % window
```

Comment:
isn't `writeInfo.SigningInfo.IndexOffset` always less than `window`?
If we're expanding from 5 -> 7, then wouldn't we just map everything from:
[0...5]
[0...5, nil, nil]
?

Reply:
no, index offset is always increasing, but then its used by modding with window size to get the new index I believe